sdk/core/azure-core-amqp/src/models/amqp_header.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#include "azure/core/amqp/models/amqp_header.hpp"
+#include "azure/core/amqp/models/amqp_value.hpp"
+
+#include <azure_uamqp_c/amqp_definitions_milliseconds.h>
+
+#include <azure_uamqp_c/amqp_definitions_header.h>
+#include <chrono>
+#include <iostream>
+
+void Azure::Core::_internal::UniqueHandleHelper<HEADER_INSTANCE_TAG>::FreeAmqpHeader(
+    HEADER_HANDLE handle)
+{
+  header_destroy(handle);
+}
+
+namespace Azure { namespace Core { namespace Amqp { namespace Models {
+  bool MessageHeader::operator==(MessageHeader const& that) const noexcept
+  {
+    return this->DeliveryCount == that.DeliveryCount && this->Durable == that.Durable
+        && this->IsFirstAcquirer == that.IsFirstAcquirer && this->Priority == that.Priority
+        && this->TimeToLive.HasValue() == that.TimeToLive.HasValue()
+        && (this->TimeToLive.HasValue() ? this->TimeToLive.Value() == that.TimeToLive.Value()
+                                        : true);
+  }
+
+  MessageHeader _internal::MessageHeaderFactory::FromUamqp(UniqueMessageHeaderHandle const& handle)
+  {
+    MessageHeader rv;
+    bool boolValue;
+    if (!header_get_durable(handle.get(), &boolValue))
+    {
+      rv.Durable = boolValue;
+    }
+
+    uint8_t uint8Value;
```

Comment:
Similar nit: Consider renaming to `priorityValue`

Reply:
uint8Value is used for all uint8 values, not just priority.